File: Lista 1/src/Craps.java
package src;
import java.util.Random;

class Craps {  
  public static void main0(String[] args) {
    Random rand = new Random();
    int dado1, dado2;
    boolean end = false;
    //estágio 1
    dado1 = rand.nextInt(5) + 1;
    dado2 = rand.nextInt(5) + 1;
    System.out.println("Dado 1: "+dado1+", Dado 2:"+dado2);
    int ponto = dado1 + dado2;
    System.out.println("Ponto: "+ponto);
    if (ponto == 11 || ponto == 7){
      System.out.println("Você Venceu, que Sortudo!!!");
      end = true;
    }
    if (ponto == 2 || ponto == 3 || ponto == 12){
      System.out.println("Você Perdeu que azar...");
      end = true;
    }
    //estagio 2
    System.out.println("Iniciado estágio 2");
    int a = 1;
    while(end == false){
      dado1 = rand.nextInt(5) + 1;
      dado2 = rand.nextInt(5) + 1;
      System.out.println("Dado 1: "+dado1+", Dado 2:"+dado2);
      System.out.println("Jogada "+a+": "+ (dado1 + dado2));
      if (ponto == dado1 + dado2){
        System.out.println("Você Venceu, que Sortudo!!!");
        end = true;
      }
      if (7 == dado1 + dado2){
        System.out.println("Você Perdeu que azar...");
        end = true;
      }
      a++;
    }
  }
}

File: Lista 2/craps.py
import random

def craps():
    #estágio 1
    dado1 = random.randint(1, 6)
    dado2 = random.randint(1, 6)
    print(f"Dado 1: {dado1}, Dado 2: {dado2}")
    ponto = dado1 + dado2
    print(f"Ponto: {ponto}")
    if ponto == 11 or ponto == 7:
        print("Você Venceu, que Sortudo!!!")
        return
    if ponto == 2 or ponto == 3 or ponto == 12:
        print("Você Perdeu que azar...")
        return
    #estagio 2
    print("Iniciado estágio 2")
    a = 1
    while True:
        dado1 = random.randint(1, 6)
        dado2 = random.randint(1, 6)
        print(f"Dado 1: {dado1}, Dado 2: {dado2}")
        print(f"Jogada {a}: {dado1 + dado2}")
        if ponto == dado1 + dado2:
            print("Você Venceu, que Sortudo!!!")
            break
        if 7 == dado1 + dado2:
            print("Você Perdeu que azar...")
            break
        a += 1

craps()